lib/opensearch/transport/client.rb

```diff
@@ -191,6 +190,32 @@ def perform_request(method, path, params = {}, body = nil, headers = nil)
         transport.perform_request(method, path, params, body, headers)
       end
 
+      def perform_ping_request(*args)
+        perform_request(*args).status == 200
+      rescue StandardError => e
+        return false if e.class.to_s =~ /NotFound|ConnectionFailed/
```

Comment:
Are we using Faraday everywhere, in which case maybe list actual class names?

Reply:
Faraday is the default library. The users can choose to use a different one. There rescues are just a simple DRY of the code segments handling delete/exists methods. Modifying them might result in unknown breaking changes